src/CodeGen_Internal.cpp

```diff
@@ -407,6 +407,16 @@ Expr lower_euclidean_mod(Expr a, Expr b) {
     }
 }
 
+Expr lower_signed_shift_left(Expr a, Expr b) {
+  Expr b_unsigned = abs(b);
```

Comment:
Might be worth adding a comment here that this is Halide's abs operator which eliminates the overflow case for the most negative value because its result is unsigned.